src/consumer/consumerGroup.js

```diff
@@ -566,4 +614,65 @@ module.exports = class ConsumerGroup {
   hasSeekOffset({ topic, partition }) {
     return this.seekOffset.has(topic, partition)
   }
+
+  /**
+   * For each of the partitions find the best nodeId to read it from
+   *
+   * @param {string} topic
+   * @param {number[]} partitions
+   * @returns {{[nodeId: number]: number[]}} per-node assignment of partitions
+   * @see Cluster~findLeaderForPartitions
+   */
+  // Invariant: The resulting object has each partition referenced exactly once
+  findReadReplicaForPartitions(topic, partitions) {
+    const partitionMetadata = this.cluster.findTopicPartitionMetadata(topic)
+    const preferredReadReplicas = this.preferredReadReplicasPerTopicPartition[topic]
+    return partitions.reduce((result, id) => {
+      const partitionId = parseInt(id, 10)
+      const metadata = partitionMetadata.find(p => p.partitionId === partitionId)
+      if (!metadata) {
+        return result
+      }
+
+      if (metadata.leader === null || metadata.leader === undefined) {
```

Comment:
This short-hand is equivalent:

```suggestion
      if (metadata.leader == null) {
```